File: backend/app/languages.py
"""Supported UI languages for ULTIMATE accounts."""

from dataclasses import dataclass


@dataclass(frozen=True)
class LanguageInfo:
    code: str
    label: str


SUPPORTED_LANGUAGES: dict[str, LanguageInfo] = {
    "fr": LanguageInfo("fr", "Français"),
    "en": LanguageInfo("en", "English"),
    "es": LanguageInfo("es", "Español"),
    "pt": LanguageInfo("pt", "Português"),
}

DEFAULT_LANGUAGE = "fr"


def is_supported_language(code: str) -> bool:
    return code.lower() in SUPPORTED_LANGUAGES

File: frontend/src/lib/languages.ts
export type PreferredLanguage = "fr" | "en" | "es" | "pt";

export const SUPPORTED_LANGUAGES: { code: PreferredLanguage; label: string }[] = [
  { code: "fr", label: "Français" },
  { code: "en", label: "English" },
  { code: "es", label: "Español" },
  { code: "pt", label: "Português" },
];

export function normalizeDiscoveryMode(mode: string | null | undefined): "near_me" | "worldwide" | "travel" {
  if (mode === "international" || mode === "worldwide") return "worldwide";
  if (mode === "travel") return "travel";
  return "near_me";
}
